### PR TITLE
Add bulk create of submission reviews

### PR DESCRIPTION
**Pull request by moshthepitt**
Fix: #1471

### Fields

- id: the id of the submission review (required when doing an update)
- instance: the id of the Instance object being reviewed
- note: the submission review comment
- status: the submission review status.  Should be one of:
  - '1': approved
  - '2': rejected
  - '3': pending

### Sample command

Send a list of objects that relate to individual submission reviews

```console
curl -X POST -H "Content-type:application/json" -d '{"status":"1","instance":1337,"note":"This was approved because it is awesome!"}, {"status":"1","instance":7331,"note":"This was approved because it is awesome!"}' https://example.com/api/v1/submissionreview/
```

### CODE REVIEW

File: onadata/apps/api/tests/viewsets/test_submission_review_viewset.py [hunk 1]
```diff
@@ -59,6 +59,68 @@ def test_submission_review_create(self):
...
+        self.extra['format'] = 'json'
+
+        request = self.factory.post('/', data=submission_data, **self.extra)
+        response = view(request=request)
+
+        self.assertEqual(201, response.status_code)
+        self.assertEqual(4, len(response.data))
+        already_seen = []
+        for item in response.data:
+            # the note sjould match what we provided
```
Comment (ivermac) [inline onadata/apps/api/tests/viewsets/test_submission_review_viewset.py]: change `sjould` to `should`

File: onadata/apps/api/tests/viewsets/test_submission_review_viewset.py [hunk 2]
```diff
@@ -59,6 +59,68 @@ def test_submission_review_create(self):
...
+            self.assertEqual('This is not very good, is it?', item['note'])
+            # the status should be rejected
+            self.assertEqual(SubmissionReview.REJECTED, item['status'])
+            # the instance id must be valid
+            self.assertTrue(instances.filter(id=item['instance']).exists())
+            # all the submission reviews must have different isntance fields
```
Comment (ivermac) [inline onadata/apps/api/tests/viewsets/test_submission_review_viewset.py]: change `isntance` to `instance`

File: onadata/apps/api/permissions.py [hunk 3]
```diff
@@ -170,6 +170,20 @@ def has_permission(self, request, view):
...
 
         if is_authenticated and view.action == 'create':
+
+            # Handle bulk create
+            # if doing a bulk create we will fail the entire process if the
+            # user lacks permissions for even one instance
+            if isinstance(request.data, list):
+                instance_ids = list(set([_['instance'] for _ in request.data]))
+                xforms = XForm.objects.filter(
+                    instances__in=instance_ids).distinct()
```
Comment (ukanga) [inline onadata/apps/api/permissions.py]: Could we avoid the join query and qury instance model directly? I suspect `has_perm()` is only going to require the `xform_id` and not the whole xform with XML + JSON data.

```python
submissions = Instance.objects.filter(id__in=instance_ids).only('xform').distinct():
for submission in submissions:
    if not request.user.has_perm(CAN_CHANGE_XFORM, xform):
...
```

**Comment by moshthepitt** [comment]
@ekisa @ukanga I have addressed your comments.  Thanks.

**Pull request merged**